src/wp-includes/option.php

```diff
@@ -1133,6 +1141,70 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ * Determine the appropriate autoload value based on input.
+ *
+ * This function checks the provided autoload value and returns a standardized
+ * value ('yes', 'no', 'default-yes', or 'default-no') based on specific conditions.
+ *
+ * @since 6.4.0
+ *
+ * @param mixed $autoload The autoload value to evaluate.
+ * @return string The standardized autoload value: 'yes', 'no', 'default-yes', or 'default-no'.
+ */
+function sanitize_autoload( $autoload ) {
+	// Check if autoload is a boolean.
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'yes' : 'no';
+	}
+
+	switch ( $autoload ) {
+		case 'no':
+		case 'yes':
+		case 'default-yes':
+		case 'default-no':
+			return $autoload;
+		default:
+			return 'default-yes';
+	}
+}
+
+/**
+ * Check the size of an option value and compare it to the maximum allowed size.
+ *
+ * This function serializes the given value and checks its size against the maximum allowed
+ * option size defined by the 'max_option_size' filter.
+ *
+ * @since 6.4.0
+ * @access private
+ *
+ * @param string $option The name of the option.
+ * @param mixed  $value  The value of the option to be checked.
+ * @return string Returns 'default-no' if the size exceeds the maximum allowed size, or 'default-yes' otherwise.
+ */
+function sanitize_option_size( $option, $value ) {
```

Comment:
IMHO this function needs a better name & description to better reflect what it actually does.

This does not "sanitize" anything, but returns the suggested autoload value for an option based on its size. Something like `determine_option_autoload_value` would be more accurate.

Reply:
That said, can we merge this function with the (also badly name `sanitize_autoload()` function above?

It could then take three arguments: `$option, $value, $autoload`

Reply:
I agree to the function rename
How should we use the third argument

Reply:
If the two functions were merged, then it would be used the same way `sanitize_autoload` uses it right now, no?

Reply:
@swissspidy I have merged the function, but this has caused the test to fail as the size no overrides the value passed in

Is this correct?

if so I will update the tests

Reply:
Looks like you changed the logic now. Previously, the size check only happened if `$autoload === null`. If you want to retain that logic, you probably need something like this (untested, unlinted):

```php
function determine_option_autoload_value( $option, $value, $autoload ) {
	// Serialize the value and check its size against the maximum allowed option size.
	$serialized_value = maybe_serialize( $value );
	$size             = strlen( $serialized_value );

	/**
	 * Filters the maximum size of option value in bytes.
	 *
	 * @since 6.4.0
	 *
	 * @param int    $max_option_size The option-size threshold, in bytes. Default 150000.
	 * @param string $option          The name of the option.
	 * @param mixed  $value           The value of the option.
	 */
	$max_option_size = (int) apply_filters( 'max_option_size', 150000, $option, $value );

	// Check if autoload is a boolean.
	if ( is_bool( $autoload ) ) {
		return $autoload ? 'yes' : 'no';
	}

	switch ( $autoload ) {
		case 'no':
		case 'yes':
		case 'default-yes':
		case 'default-no':
			return $autoload;
		default:
			return $size > $max_option_size ? 'default-no' : 'default-yes;
	}
}
```

Reply:
Made changes and the tests now pass